gitattributes.go

```diff
@@ -0,0 +1,251 @@
+package enry
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"regexp"
+	"strings"
+
+	"gopkg.in/src-d/enry.v1/data"
+)
+
+type attrType int
+
+const (
+	vendor attrType = iota
+	documentation
+	generated
+	language
+)
+
+const _attrType_name = "vendordocumentationgeneratedlanguage"
+
+var _attrType_index = [...]uint8{0, 6, 19, 28, 36}
+
+func (i attrType) String() string {
+	if i < 0 || i >= attrType(len(_attrType_index)-1) {
+		return fmt.Sprintf("attrType(%d)", i)
+	}
+	return _attrType_name[_attrType_index[i]:_attrType_index[i+1]]
+}
+
+type boolAttribute struct {
+	kind       attrType
+	matchers   []string
+	attributes map[string]bool
+}
+
+type regExpAttribute struct {
+	kind       attrType
+	matchers   []string
+	attributes map[*regexp.Regexp]string
+}
+
+// GitAttributes is a struct that contains two maps, boolAttributes contains all the attributes that works like a boolean condition,
+// regExpAttributes contains all the attributes that match a regExp to choose if an attribute is applied or not
+type GitAttributes struct {
+	boolAttributes   map[attrType]boolAttribute
+	regExpAttributes map[attrType]regExpAttribute
+}
+
+type overrideError struct {
+	attribute attrType
+	path      string
+}
+
+func (e *overrideError) Error() string {
+	return fmt.Sprintf("gitattributes: You are overriding a %v attribute of one of your previous lines %s\n", e.attribute, e.path)
+}
+
+// IsVendor returns whether or not path is a vendor path.
+func (gitAttrs *GitAttributes) IsVendor(path string) bool {
+	if val, ok := gitAttrs.boolAttributes[vendor].attributes[path]; ok {
+		return val
+	}
+
+	return data.VendorMatchers.Match(path)
+}
+
+// IsVendor returns whether or not path is a documentation path.
+func (gitAttrs *GitAttributes) IsDocumentation(path string) bool {
+	if val, ok := gitAttrs.boolAttributes[documentation].attributes[path]; ok {
```

Comment:
Maybe you can keep only those attributes which are "true". If the attribute is not in the map it's false, so you can check if the attribute is in the map.
`
if _, ok := gitAttrs.boolAttributes[documentation].attributes[path]; ok {
        return ok
}
`

In this way, if we eventually allow attributes overriding, when load an attribute set to false, look for it in the map to delete it